Core/include/Acts/Seeding/SeedFilter.ipp

```diff
@@ -65,34 +65,44 @@ void SeedFilter<external_spacepoint_t>::filterSeeds_2SpFixed(
               });
   }
 
-  for (auto& i : idx) {
+  size_t startCompTopLoopIdx = 0;
+  // loop over top SPs and other compatible top SP candidates
+  for (auto& topSPIdx : idx) {
     // if two compatible seeds with high distance in r are found, compatible
     // seeds span 5 layers
     // -> weaker requirement for a good seed
     std::vector<float> compatibleSeedR;
 
-    float invHelixDiameter = invHelixDiameterVec[i];
+    float invHelixDiameter = invHelixDiameterVec[topSPIdx];
     float lowerLimitCurv = invHelixDiameter - m_cfg.deltaInvHelixDiameter;
     float upperLimitCurv = invHelixDiameter + m_cfg.deltaInvHelixDiameter;
     // use deltaR instead of top radius
-    float currentTop_r = m_cfg.useDeltaRorTopRadius ? topSpVec[i]->deltaR()
-                                                    : topSpVec[i]->radius();
-    float impact = impactParametersVec[i];
+    float currentTop_r = m_cfg.useDeltaRorTopRadius
+                             ? topSpVec[topSPIdx]->deltaR()
+                             : topSpVec[topSPIdx]->radius();
+    float impact = impactParametersVec[topSPIdx];
 
     float weight = -(impact * m_cfg.impactWeightFactor);
-    for (auto& j : idx) {
-      if (i == j) {
+
+    for (size_t j = startCompTopLoopIdx; j < idx.size(); j++) {
+      size_t compatibleTopSPIdx = idx[j];
+      if (compatibleTopSPIdx == topSPIdx) {
         continue;
       }
 
-      float otherTop_r = m_cfg.useDeltaRorTopRadius ? topSpVec[j]->deltaR()
-                                                    : topSpVec[j]->radius();
+      float otherTop_r = m_cfg.useDeltaRorTopRadius
```

Comment:
Variable naming comment, camelCase style is used in this code, so maybe `otherTopR` ?